Analysis/output/model_failures_report.Rmd

```diff
@@ -0,0 +1,177 @@
+---
+title: "Model Failures Report"
+output: html_document
+params:
+  cholera_directory: "~/gh/cholera-mapping-pipeline/"
+  config_directory: "Analysis/configs/2015_2019_full_base/"
+---
+```{r setup, include=FALSE, dev="CairoPNG"}
+knitr::opts_chunk$set(
+  echo = FALSE, 
+  dev="CairoPNG", 
+  error = TRUE,
+  fig.align = "center",
+  message = FALSE,
+  warning = FALSE,
+  fig.width = 10,
+  fig.height = 10
+  )
+library(stringr)
+library(dplyr)
+library(magrittr)
+library(tidyr)
+library(readr)
+library(ggplot2)
+library(taxdat)
+```
+
+# Check progress along mapping pipeline
+
+```{r tab_progress, fig.cap = "Model progress report"}
+
+print(params$config_directory)
+
+try({
+
+  taxdat::create_mapping_progress_table(
+    config_path = params$config_directory,
+    pipeline_path = params$cholera_directory
+  ) %>%
+    dplyr::group_by(config_path) %>%
+    dplyr::summarize(stages_complete = ifelse(
+      data,
+      ifelse(covar,
+        ifelse(stan_input,
+          ifelse(stan_output,
+            "Output Produced",
+            "Input Produced"
+          ), "Covariates constructed"
+        ),
+        "Data pulled"
+      ),
+      "None"
+    )) %>%
+    tidyr::separate(config_path,
+      c(NA, NA, "start_year","end_year", NA, NA, "country", NA) # This is file path specific
+    ) %>%
+    ggplot2::ggplot() +
+      ggplot2::geom_tile(ggplot2::aes(x=paste(start_year, end_year, sep = '-'), y=country, fill=stages_complete))
+
+  })
+
+  ```
+
+# Extract Stan outputs
+
+```{r extract_stan_output}
+
+all_configs <- list.files(stringr::str_c(params$cholera_directory, params$config_directory), full.names = TRUE)
+
+cases_check <- list()
+rhat_check <- list()
+
+for (i in 1:length(all_configs)){
+  config_filename <- all_configs[i]
+  config <- yaml::read_yaml(config_filename)
+  file_names <- taxdat::get_filenames(config, params$cholera_directory)
+  cntry_code <- stringr::str_remove(unlist(strsplit(config_filename, "/"))[length(unlist(strsplit(config_filename, "/")))], ".yml")
+  print(paste("Reading files associated with", config_filename))
+
+  ## get cases
+  try({
+    model.rand <- taxdat::read_file_of_type(file_names[["stan_output"]], "model.rand")
```

Comment:
similarly, we should use the same file_names conventions across reports i think

Reply:
The model.rand notation is consistent over all of the reports I've seen